engine/apps/phone_notifications/exceptions.py

```diff
@@ -1,24 +1,40 @@
-class FailedToMakeCall(Exception):
+class BaseFailed(Exception):
+    """
+    Failed is base exception for all Failed... exceptions.
+    This exception is indicates error while performing some phone notification operation.
+    Optionally can contain graceful_msg attribute. When graceful_msg is provided it mean that error on provider side is
+    not our fault, but some provider error (number is blocked, fraud guard, ...).
+    By default, graceful_msg is None - it means that error is our fault (network problems, invalid configuration,...).
+
+    Attributes:
+       graceful_msg: string with some details about exception which can be exposed to caller.
+    """
+
+    def __init__(self, graceful_msg=None):
```

Comment:
where is `graceful_msg` being set?

Reply:
nowhere yet. but it's expected to be set in concrete providers, me or @mderynck will add this

Reply:
nowhere yet. but it's expected to be set in concrete providers, me or @mderynck will add this